ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/validator/SubmissionValidator.scala

```diff
@@ -40,7 +41,9 @@ class SubmissionValidator[LogResult](
     ) => LogEntryAndState,
     allocateLogEntryId: () => DamlLogEntryId,
     checkForMissingInputs: Boolean = false,
-)(implicit executionContext: ExecutionContext) {
+)(implicit executionContext: ExecutionContext, logCtx: LoggingContext) {
```

Comment:
@miklos-da how public of an API is the SubmissionValidator? Will we use this directly or still wrap it with something? If it's the public API for integrations then isn't it a problem to require use of our internal logging here?

Reply:
Good point. We can construct the logging context inside the `SubmissionValidator` if necessary to hide it from API consumers; this is basically the same as using slf4j directly.

Reply:
`ValidatingCommitter` also exposes a dependency on `LoggingContext`. I'm going to remove it from the interface but still pass it through to the `SubmissionValidator.validateAndCommit` via a package-private method called `SubmissionValidator.validateAndCommitWithLoggingContext`. Does this make sense to do?

Reply:
Yes, that seems like a good approach. `SubmissionValidator` is on the public API hence we should avoid requiring a `LoggingContext` in order to use it.